pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactorTool.java

```diff
@@ -0,0 +1,140 @@
+
+        @Parameter(names = {"-C", "--client-conf"}, description = "Configuration file for Pulsar Client")
+        private String clientConfigFile = Paths.get("").toAbsolutePath().normalize().toString() + "/conf/client.conf";
+
+        @Parameter(names = {"-t", "--topic"}, description = "Topic to compact", required=true)
+        private String topic;
+
+        @Parameter(names = {"-h", "--help"}, description = "Show this help message")
+        private boolean help = false;
+    }
+
+    public static void main(String[] args) throws Exception {
+        Arguments arguments = new Arguments();
+        JCommander jcommander = new JCommander(arguments);
+        jcommander.setProgramName("PulsarTopicCompactor");
+
+        // parse args by JCommander
+        jcommander.parse(args);
+        if (arguments.help) {
+            jcommander.usage();
+            System.exit(-1);
+        }
+
+        // init broker config
+        ServiceConfiguration brokerConfig;
+        if (isBlank(arguments.brokerConfigFile)) {
+            jcommander.usage();
+            throw new IllegalArgumentException("Need to specify a configuration file for broker");
+        } else {
+            brokerConfig = PulsarConfigurationLoader.create(
+                    arguments.brokerConfigFile, ServiceConfiguration.class);
+        }
+
+        String pulsarServiceUrl = PulsarService.brokerUrl(brokerConfig);
+        ClientConfiguration clientConfig = new ClientConfiguration();
+
+        if (!isNotBlank(arguments.clientConfigFile)) {
+            try (FileInputStream fis = new FileInputStream(arguments.clientConfigFile)) {
+                Properties properties = new Properties();
+                properties.load(fis);
+
+                if (isNotBlank(properties.getProperty("serviceUrl"))) {
+                    pulsarServiceUrl = properties.getProperty("serviceUrl");
+                }
+                if (isNotBlank(properties.getProperty("authPlugin"))) {
+                    clientConfig.setAuthentication(properties.getProperty("authPlugin"),
+                                                   properties.getProperty("authParams"));
+                }
+
+                clientConfig.setUseTls(Boolean.parseBoolean(properties.getProperty("useTls")));
+                clientConfig.setTlsAllowInsecureConnection(
+                        Boolean.parseBoolean(properties.getProperty("tlsAllowInsecureConnection")));
+                clientConfig.setTlsTrustCertsFilePath(properties.getProperty("tlsTrustCertsFilePath"));
+            }
+        }
+
+        ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder().setNameFormat("compaction-%d").setDaemon(true).build());
+
+        ZooKeeperClientFactory zkClientFactory = new ZookeeperBkClientFactoryImpl();
+
+        ZooKeeper zk = zkClientFactory.create(brokerConfig.getZookeeperServers(),
```

Comment:
We shouldn't need the broker configuration to get hold of the ZK connection string. We can use the new API to get this config from the brokers. See https://github.com/apache/incubator-pulsar/pull/1223